content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+
+```yaml
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - file
+      - bison
+      - libtool
+      - ca-certificates-bundle
+      - bzip2-dev
+      - libxml2-dev
+      - curl-dev
+      - openssl-dev
+      - readline-dev
+      - sqlite-dev
+      - libsodium-dev
+      - libpng-dev
+      - libjpeg-turbo-dev
+      - libavif-dev
+      - libwebp-dev
+      - libxpm-dev
+      - libx11-dev
+      - freetype-dev
+      - gmp-dev
+      - icu-dev
+      - openldap-dev
+      - oniguruma-dev
+      - libxslt-dev
+      - postgresql-15-dev
+      - libzip
+```
+
+Don't worry if you don't know everything you'll need upfront at build time. Even if you build the package locally first, your system most likely has many dependencies already installed; by paying attention to the output provided by melange, you will be able to figure out what is missing, and iterate until your build environment looks right.
+
+One thing that may happen during this process is finding out that one or more dependencies needed by your package are not yet available in Wolfi, so they need to be built first. It is a normal part of the process, so don't worry – you will be able to build incrementally and test everything locally.
+
+### The `pipeline` Section
+
+With package metadata and build environment defined, it's time to create the pipeline that will build your package. The pipeline section looks a lot like a GitHub Actions workflow, defining a series of steps that must be executed in the same order they are defined, creating output that will be packaged into one or more apk packages.
+
+A package build pipeline typically starts with fetching the package (as a tarball or directly from a Git branch) and matching the downloaded artifact against an expected sha hash.
+
+Some of the actions executed in build pipelines are very similar across packages: downloading a package, running configure and make, fetching a package from git… Luckily for us, melange bakes a lot of repetitive tasks into reusable [pipelines](https://github.com/chainguard-dev/melange/tree/main/pkg/build/pipelines):
+
+- Downloading Packages
+  - `fetch`
+  - `git-checkout`
+- Autoconf
+  - `autoconf/configure`
+  - `autoconf/make`
+  - `autoconf/make-install`
+- Cmake
+  - `cmake/build`
+  - `cmake/configure`
+  - `cmake/install`
+- Go
+  - `go/build`
+  - `go/install`
+- Meson
+  - `meson/compile`
+  - `meson/configure`
+  - `meson/install`
+- Ruby
+  - `ruby/build`
+  - `ruby/clean`
+  - `ruby/install`
+- Split
+  - `split/debug`
+  - `split/dev`
+  - `split/infodir`
+  - `split/locales`
+  - `split/manpages`
+  - `split/static`
+- Other
+  - `strip`
+  - `patch`
+
+Each pipeline can have one or more parameters that should be provided as keypairs in a `with` entry. For example, this is how a download-and-check step looks like in the melange YAML, using the built-in pipeline `fetch`:
```

Comment:
```suggestion
Each pipeline can have one or more parameters that should be provided as keypairs in a `with` entry. For example, a download-and-check has the following structure in the melange YAML, using the built-in pipeline `fetch`:
```

Trying to remove words like "look" and "see" when I find them

Reply:
thank you!!!